### PR TITLE
fix(FOROME-1274): saving preset on ws-page after reloading

### PR DESCRIPTION
**Pull request by Lapk1n**
https://quantori.atlassian.net/browse/FOROME-1274

### CODE REVIEW

**Review by QSdmitrioul — APPROVED** [review]

File: src/pages/filter/refiner/refiner.page.tsx [hunk 1]
```diff
@@ -18,25 +17,16 @@ import { FilterControl } from '@pages/filter/common/filter-control/filter-contro
...
   useDatasetName()
 
-  useEffect(() => {
-    presetName && applyPreset(presetName)
-
-    return () => {
-      dirInfoStore.resetData()
-    }
-  }, [presetName])
+  usePreset()
```
Comment (AndreyUstyumenko) [inline src/pages/filter/refiner/refiner.page.tsx]: Make it more generic to use it on dtree as well then

File: src/core/hooks/use-preset.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+export const usePreset = () => {
+  const params = useParams()
+  const presetName = params.get('preset') || ''
+
+  useEffect(() => {
+    presetName && applyPreset(presetName)
+
+    return () => {
+      dirInfoStore.resetData()
```
Comment (AlMaQntr) [inline src/core/hooks/use-preset.ts]: Why we need to reset dirInfoStore data when applying preset?

File: src/core/hooks/use-preset.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { useEffect } from 'react'
+
+import dirInfoStore from '@store/dirinfo'
+import { applyPreset } from '@pages/filter/refiner/components/solution-control-refiner/solution-control-refiner.utils'
```
Comment (AlMaQntr) [inline src/core/hooks/use-preset.ts]: i think, we should do it in store level, preset store should observe urls and apply preset from it. I working on this base functionality now. Maybe we can wait a bit with this change

**Review by AlMaQntr — COMMENTED** [review]
Please wait for generic history observer for stores

File: src/core/hooks/use-solution.ts [hunk 4]
```diff
@@ -6,25 +6,14 @@ import datasetStore from '@store/dataset/dataset'
...
   const solutionName = params.get(solutionType) || ''
 
   useEffect(() => {
-    if (solutionName) {
-      onApply(solutionName)
-    }
-
     if (solutionType === SolutionTypesEnum.Preset) {
       if (activeSolution && !solutionName) {
         pushQueryParams({ preset: activeSolution })
```
Comment (AndreyUstyumenko) [inline src/core/hooks/use-solution.ts]: shouldn't it be done by observeHistory?

Comment (Lapk1n) [reply src/core/hooks/use-solution.ts]: I thought a history observer couldn't do that, but correct me if I'm wrong. @AlMaQntr

**Review by AndreyUstyumenko — APPROVED** [review]

**Pull request merged**